### DISCUSSION TITLE
500 Error when pointing to the same data object

### DISCUSSION THREAD

**Discussion by ghost** [spatie/laravel-data#61] [Q&A] [answered]
In one of our projects we have a **WorkerData**, inside that **WorkerData** we Lazy the **managers** of a **worker**, managers is a **hasManyThrough()** from a join table which gives the **Worker** model again, the problem is, when validating the data (PUT/POST), it throws a 500 error `Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes)`, because it points to the same data object again, to solve the problem I could create a new WorkerManagerData object but would have to add the same properties again.

Example:
```php
<?php

class WorkerData extends Data
{
    public function __construct(
       public string $first_name,
       public string $last_name,

       #[DataCollectionOf(WorkerData::class)]
       public DataCollection | Lazy $managers,
    ) {}

    public static function fromModel(Worker $worker): self
    {
        return new self(
            managers: Lazy::whenLoaded('managers', $worker, fn () => WorkerData::collection($worker->managers)),
        );
    }
    ...
```

Also tested with an hasOne() relation and the same issue happens, for example if the worker only had 1 manager:

```php
<?php

class WorkerData extends Data
{
    public function __construct(
       public string $first_name,
       public string $last_name,

       public WorkerData | Lazy $manager,
    ) {}

    public static function fromModel(Worker $worker): self
    {
        return new self(
            manager: Lazy::whenLoaded('manager', $worker, fn () => WorkerData::fromModel($worker->manager)),
        );
    }
    ...
```

**Comment by rubenvanassche** [comment, marked as answer]
Hi @mazzv1,

This is because the package will automatically try to deduce validation rules for the object. Since the `WorkerData` keeps getting nested, validation rules are also getting nested which causes an infinite loop.

You can fix this by setting the rules manually for the `manager` property using the `rules` method.

Be sure to do this on version 1.3.0, because on previous versions the memory bug would still occur.

**Reply by roykesserwani** [reply]
Can you write an example for this? I tried that but it didn't work. 

@rubenvanassche

**Reply by ghost** [reply]
Hello @roykesserwani, got around this by doing what @rubenvanassche said, adding the rules() method on the WorkerData fixed the problem.

```php
public static function rules(): array
{
        return [
            'managers' => [],
        ];
}
```

**Reply by roykesserwani** [reply]
thank you! much appreciated @mazzv1
